src/rtt.c

```diff
@@ -120,24 +121,16 @@ uint32_t memsrch(target *cur_target)
 	if (srch_str_len == 0 || srch_str_len > sizeof(srch_buf) / 2)
 		return 0;
 
-	if (rtt_cbaddr && !target_mem_read(cur_target, srch_buf, rtt_cbaddr, srch_str_len)
-		&& strncmp((const char *)(srch_buf), srch_str, srch_str_len) == 0)
-		/* still at same place */
-		return rtt_cbaddr;
-
-	for (struct target_ram *r = cur_target->ram; r; r = r->next) {
-		uint32_t ram_end = r->start + r->length;
-		for (uint32_t addr = r->start; addr < ram_end; addr += sizeof(srch_buf) - srch_str_len - 1) {
-			uint32_t buf_siz = MIN(ram_end - addr, sizeof(srch_buf));
-			if (target_mem_read(cur_target, srch_buf, addr, buf_siz)) {
-				gdb_outf("rtt: read fail at 0x%" PRIx32 "\r\n", addr);
-				continue;
-			}
-			for (uint32_t offset = 0; offset + srch_str_len + 1 < buf_siz; offset++) {
-				if (strncmp((const char *)(srch_buf + offset), srch_str, srch_str_len) == 0) {
-					uint32_t cb_addr = addr + offset;
-					return cb_addr;
-				}
+	for (uint32_t addr = ram_start; addr < ram_end; addr += sizeof(srch_buf) - srch_str_len - 1) {
+		uint32_t buf_siz = MIN(ram_end - addr, sizeof(srch_buf));
+		if (target_mem_read(cur_target, srch_buf, addr, buf_siz)) {
+			gdb_outf("rtt: read fail at 0x%" PRIx32 "\r\n", addr);
+			continue;
+		}
+		for (uint32_t offset = 0; offset + srch_str_len + 1 < buf_siz; offset++) {
+			if (strncmp((const char *)(srch_buf + offset), srch_str, srch_str_len) == 0) {
+				uint32_t cb_addr = addr + offset;
```

Comment:
`const` or perhaps even merge the calculation into the return statement, dropping the {}'s for the if block?